### PR TITLE
Simplify reconnect logic, handle ZK errors

### PR DESCRIPTION
**Pull request by dszoboszlay**
My final PR for now.
* Simplify reconnect logic.
* Make client more robust by applying flow control and handling ZK errors.
* Maintenance API to update ZK server list (and as a bonus: also perform DNS lookup on ZK hosts every time we attempt to reconnect instead of only once upfront).
* Add Klarna copyright and point links to this repo instead of upstream.

### CODE REVIEW

**Review by mikpe — APPROVED** [review]

File: package.exs [hunk 1]
```diff
@@ -3,15 +3,15 @@ defmodule Erlzk.MixFile do
 
   def project do
     [app: :erlzk,
-     version: "0.6.5",
+     version: "0.6.5+klarna1",
```
Comment (zmstone) [inline package.exs]: Non-semver version will cause trouble for hex package users.

Delete this file if we are not going to publish it to hex.

if we are, then:
we can switch to use rebar3 for hex package management then there is no dependency to elixir.
we do not have permission to publish to erlzk repo in hex.
so we'll need to use another name like the other two listed here:
https://hex.pm/packages?_utf8=%E2%9C%93&search=erlzk&sort=recent_downloads

Comment (dszoboszlay) [reply package.exs]: Thanks, I wasn't aware! I think I'll just drop this file, I have no intention in publishing this to hex.

**Review by zmstone — APPROVED** [review]

File: src/erlzk_conn.erl [hunk 2]
```diff
@@ -50,10 +51,10 @@
     socket,
     host,
     port,
-    proto_ver,
+    proto_ver = 0,
     timeout,
-    session_id,
-    password,
+    session_id = 0,
+    password = <<0:128>>,
```
Comment (k32) [inline src/erlzk_conn.erl:57]: Shouldn't we hide secrets in a closure?

Comment (k32) [reply src/erlzk_conn.erl:57]: :snail:

Comment (dszoboszlay) [reply src/erlzk_conn.erl:57]: Good idea, but at the same time, the connection to ZK is plain text... :rofl: :cry:
I can fix this in a follow up PR though.

Comment (k32) [reply src/erlzk_conn.erl:57]: > the connection to ZK is plain text

God help us all... Then it doesn't make much sense

Comment (zmstone) [reply src/erlzk_conn.erl:57]: as far as I know TLS is still in beta.
we probably don't want to run a beta version.

Comment (id) [reply src/erlzk_conn.erl:57]: TLS is not critical for our use case, we can to live without it and enable as soon as it's available in a stable version.

**Pull request merged**